### ISSUE TITLE
New rule proposal: A "for" loop update clause should move the counter in the right direction

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#8387]
**Please describe what the rule should do:**

A for loop with a stop condition that can never be reached, such as one with a counter that moves in the wrong direction, will run infinitely. While there are occasions when an infinite loop is intended, the convention is to construct such loops as while loops. More typically, an infinite for loop is a bug.

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error

**Provide 2-3 code examples that this rule will warn about:**

```js
// bad. it will never end.
for (var i = 0; i < 5; i--) {
  // do something.
}
```
```js
// bad. it will never end.
for (var i = 5; i >= 0; i++) {
  // do something.
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

This rule seems like a fairly universal potential mistake. It meets all the core rule guidelines and is a perfect candidate for linting.

**Comment by not-an-aardvark** [comment]
Thanks for the rule suggestion. This seems like a good idea to me. :+1:

**Comment by platinumazure** [comment]
I'll champion, and now we can accept this. Thanks for your patience @Aladdin-ADD!
